### PR TITLE
Groups page

### PR DESCRIPTION
**Pull request by Jawayria**
Groups Page

### CODE REVIEW

File: cric-predict/src/Components/CreateGroup.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import '../Stylesheets/App.css';
+import axios from 'axios';
+import {Button, Modal, Form, Row, Col}  from 'react-bootstrap';
+
+
+export default class CreateGroupsComponent extends React.Component  {
+
+        constructor(){
+            super();
```
Comment (mraarif) [inline cric-predict/src/Components/CreateGroup.js:10]: can't we omit `super()` call here?

File: cric-predict/src/Components/CreateGroup.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            this.state = {
+                name: '',
+                privacy: 'public',
+                all_users:[],
+                selected_users: [],
+                filtered_users:[],
+                users_copy: [],
+                showHide : false
```
Comment (mraarif) [inline cric-predict/src/Components/CreateGroup.js:18]: why not name it `showModal` so it conveys clearly what it's doing.

File: cric-predict/src/Components/CreateGroup.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        }
+
+        async componentDidMount() {
+            const response = await axios.get('http://localhost:8000/api/user/list/',{
+            headers: {
+                'Authorization': "Bearer "+window.localStorage.getItem('access_token')
+            }
+            })
+            const users_list = response.data
+            this.setState({all_users: users_list, filtered_users:users_list, users_copy: users_list})
```
Comment (mraarif) [inline cric-predict/src/Components/CreateGroup.js]: why do we need to make a copy of users? if it's necessary we should have a better name for it and use the spread operator to make a copy of an array (preferred way to make copies of reference types in ES6)

File: cric-predict/src/Components/CreateGroup.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        handleChange = (event) => {
+            this.setState({[event.target.name]: event.target.value, [event.target.privacy]: event.target.value});
+        };
+
+        handleSubmit = async (event) => {
+            event.preventDefault();
+            await axios.post('http://localhost:8000/api/group/', {"name": this.state.name, "privacy": this.state.privacy,
+                        "users": this.state.selected_users},{
+                        headers: {
+                            'Authorization': "Bearer "+window.localStorage.getItem('access_token')
```
Comment (mraarif) [inline cric-predict/src/Components/CreateGroup.js]: can't we use `localStorage` directly?

File: cric-predict/src/Components/Groups.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import '../Stylesheets/App.css';
+import '../Stylesheets/Group.css';
+import {Button}  from 'react-bootstrap';
+import PublicGroups from './PublicGroups';
+import UserGroups from './UserGroups';
+import CreateGroup from './CreateGroup';
+import {Redirect} from "react-router-dom";
+
+function GroupComponent() {
+      if(!window.localStorage.getItem('access_token')) {
```
Comment (mraarif) [inline cric-predict/src/Components/Groups.js]: maybe we can explore a third-party plugin that will handle authentication and redirection on our behalf

File: cric-predict/src/Components/Login.js [hunk 6]
```diff
@@ -2,34 +2,44 @@ import React from 'react';
...
 import axios from 'axios';
+import {Redirect} from "react-router-dom";
 
 
 class LoginComponent extends React.Component {
 
     constructor(props) {
         super(props);
-        this.state = {username: '', password: ''};
+        this.state = {username: '', password: '', loggedin: false};
```
Comment (mraarif) [inline cric-predict/src/Components/Login.js:12]: what would happen if the user is logged in and lands on the login page? he/she would need to log in again?

File: cric-predict/src/Components/PublicGroups.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import '../Stylesheets/App.css';
+import axios from 'axios';
+import {Button, Modal, Form, Row, Col}  from 'react-bootstrap';
+
+
+export default class PublicGroupsComponent extends React.Component  {
```
Comment (mraarif) [inline cric-predict/src/Components/PublicGroups.js]: can't we just handle this in one component? I'm assuming we've two different components with almost same rendering logic

File: cric-predict/src/Components/Signup.js [hunk 8]
```diff
@@ -2,13 +2,14 @@ import React from 'react';
...
 import axios from 'axios';
+import {Redirect} from "react-router-dom";
 
 
 class SignupComponent extends React.Component {
 
     constructor(props) {
         super(props);
-        this.state = {username: '', password: '', password2: ''};
+        this.state = {username: '', password: '', password2: '', loggedin:false};
```
Comment (mraarif) [inline cric-predict/src/Components/Signup.js:12]: same comment here as on Login component

File: cric-predict/src/Components/UserGroups.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    filterGroupList = async (event) => {
+            this.setState({filtered_groups:this.state.groups.filter(group => group.name.toLowerCase().includes(event.target.value.toLowerCase()))});
+        }
+
+
+  render() {
+  return (
+        <div className="card">
+            <div className="card-body">
+                <h3 className="text-style"> Your Groups</h3>
```
Comment (mraarif) [inline cric-predict/src/Components/UserGroups.js]: we should be able to handle this in one component by aggregating data on the backend ... let me know if you need help with this

**Review by mraarif — COMMENTED** [review]
I've added a few comments

File: cric-predict/src/Components/CreateGroup.js (hunk 1 above)
Comment (Jawayria) [reply cric-predict/src/Components/CreateGroup.js:10]: I can't use 'this' if I omit 'super()' so I have to mention it

File: cric-predict/src/Components/CreateGroup.js (hunk 2 above)
Comment (Jawayria) [reply cric-predict/src/Components/CreateGroup.js:18]: I guess, I may have to show a modal in other components as well later on so I specified that this one is to create group

File: cric-predict/src/Components/CreateGroup.js (hunk 4 above)
Comment (Jawayria) [reply cric-predict/src/Components/CreateGroup.js]: Yes we can.

File: cric-predict/src/Components/Groups.js (hunk 5 above)
Comment (Jawayria) [reply cric-predict/src/Components/Groups.js]: Okay

File: cric-predict/src/Components/Login.js (hunk 6 above)
Comment (Jawayria) [reply cric-predict/src/Components/Login.js:12]: Well, I guess I will have to add some checks for the access token before displaying login page

**Review by mraarif — APPROVED** [review]

**Pull request merged**